### DISCUSSION TITLE
How to check that R.string is not used

### DISCUSSION THREAD

**Discussion by ravenfeld** [LemonAppDev/konsist#1623] [Ask a Question]
### Wanted end result.

Hello,

I am trying to verify that in my code I am not using R.string.

If I look at the import of R it induces false positives because I leave the right to use R.drawable.

Thank you in advance for your help.

### What you tried so far.

I tried to look at the imports but it doesn't allow to know how it is used afterwards.

I tried to look at the properties but I have the impression that it only comes out of my classes and not of the generated classes

**Comment by igorwojda** [comment]
Konsist is more about code structure verification of project (non-generated) code base. 

You could use `text` property from `file` as a fallback, however it may be slow, so it's worth testing the performance.

```konsist
Konsist
	.scopeFromProject()
	.files
	.assertFalse { 
		val regex = """R\.((?!drawable)[a-zA-Z]+)\.[a-zA-Z0-9_]+""".toRegex()
        it.text.contains(regex)
		// or 
        // it.text.contains("R.string")
	}
```

Regex Test cases
```
val test1 = "R.string.app_name"      // true
val test2 = "R.drawable.ic_launcher" // false
val test3 = "R.layout.activity_main" // true
val test4 = "R.id.button"           // true
```
